mlir/tools/miopen-gen/miopen-gen.cpp

```diff
@@ -1195,248 +1204,136 @@ createVerifierFunc(ModuleOp &module, const KernelIF &kernel,
-    // <test> = select (<cpu> != 0.0), <divf>, <subf>)
-    auto notZeroOp = testBody.create<arith::CmpFOp>(
-        loc, arith::CmpFPredicate::UNE, cpuLoadVal, fval00);
-    auto testOp =
-        testBody.create<arith::SelectOp>(loc, notZeroOp, divfOp, subfOp);
-    percentDiffVal = divfOp;
-
-    // <test> = |<test>|
-    auto absfOp = testBody.create<math::AbsOp>(loc, testOp);
-
-    auto absCpuVal = testBody.create<math::AbsOp>(loc, cpuLoadVal);
-
-    mlir::Value maxPercentVal;
-    if (elemType.isF16()) {
-      // The threshold for f16 datatype is controlled by -threshold
-      // The default value is set to 0.25
-      maxPercentVal = fval025;
-    } else {
-      maxPercentVal = fval000001; // 0.0001 %
-    }
-
-    // <test> >= <max_percent>
-    cmpVal = testBody.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UGT,
-                                            absfOp, maxPercentVal);
-    if (elemType.getIntOrFloatBitWidth() < 32) {
-      // && <cpu> >= 0.001f
-      auto cmp1Op = testBody.create<arith::CmpFOp>(
-          loc, arith::CmpFPredicate::UGT, absCpuVal, fval001);
-
-      cmpVal = testBody.create<arith::AndIOp>(loc, cmpVal, cmp1Op);
-
-      // TODO?: check for GPU inf
-    }
-
-  } else if (elemType.isF32()) {
-    cmpVal = loopB.create<arith::CmpFOp>(loc, arith::CmpFPredicate::UNE,
-                                         cpuLoadVal, gpuLoadVal);
-    percentDiffVal = loopB.create<arith::SubFOp>(loc, cpuLoadVal, gpuLoadVal);
-  } else {
-    cmpVal = loopB.create<arith::CmpIOp>(loc, arith::CmpIPredicate::ne,
-                                         cpuLoadVal, gpuLoadVal);
-  }
-
-  scf::IfOp ifOp = testBody.create<scf::IfOp>(loc, cmpVal, false);
-  auto thenBody = ifOp.getThenBodyBuilder();
-
-  thenBody.create<memref::StoreOp>(loc, c0ConstantInt32Op, cmpResultAllocOp,
-                                   ValueRange{c0IndexOp});
+  auto getF32Val = [&](float val) -> mlir::Value {
+    llvm::APFloat apVal(val);
+    return b.create<arith::ConstantFloatOp>(loc, apVal, floatType);
+  };
+  // Thresholds for different metrics
+  // RMS: 0.00003f for all data types
+  // absDiff: 100.0f for all data types, i.e. the maxAbsDiff metric is disabled
+  // relDiff 100.0f for f16, i.e. maxRelDiff metric is disabled for f16
+  // datatypes
+  //         0.000001f for other data types
+  auto thr_RMS = getF32Val(RMSThreshold.getValue());
+  auto thr_absDiff = getF32Val(absDiffThreshold.getValue());
+  mlir::Value thr_relDiff;
+  if (testOutType.isF16())
+    thr_relDiff = getF32Val(relDiffThreshold.getValue());
+  else
+    thr_relDiff = getF32Val(0.000001f);
 
-  // call mcpuPrintF32(float f1, float f2)
-  func::FuncOp printFunc;
-  if (elemType.isIntOrIndex()) {
-    printFunc = makeFuncDecl(module, "mcpuPrintInt32", {intType, intType});
+  // obtain function name of the verifier wrapper
+  std::string verifyFuncName = "mcpuVerify5D";
+  if (valElemType.isF32())
+    verifyFuncName += "Float";
+  else
```

Comment:
In the interests of paranoia, can we get this to be `else if .isIntegerType(32) ... else (emit error)`